### PR TITLE
Add HCR document update event.

### PR DESCRIPTION
**Pull request by yaohaizh**
@akaroml @testforstephen @andxu

### CODE REVIEW

File: src/hotCodeReplace.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        if (t === "java") {
+            runningSessions.delete(session);
+            suppressedReasons.clear();
+        }
+    }));
+
+    context.subscriptions.push(vscode.debug.onDidReceiveDebugSessionCustomEvent((customEvent) => {
+        const t = customEvent.session ? customEvent.session.type : undefined;
+        if (t === "java") {
+            if (customEvent.event === "hotCodeReplace" && !suppressedReasons.has(customEvent.body.message)) {
```
Comment (akaroml) [inline src/hotCodeReplace.ts]: The event name should be hotCodeReplaceEnd.
And we also need the hotCodeReplaceStart event to let user know the progress.

File: src/hotCodeReplace.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+const runningSessions: Set<vscode.DebugSession> = new Set();
+
+const suppressedReasons: Set<string> = new Set();
+
+const NOT_SHOW_AGAIN: string = "Not show again";
+
+const JAVA_LANGID: string = "java";
+
+const HCR_EVENT = "hotCodeReplace";
```
Comment (akaroml) [inline src/hotCodeReplace.ts]: I think we need two events: hotCodeReplaceStart and hotCodeReplaceEnd. And for any event, we need the error object to see whether it's a successful one.

hotCodeReplaceStart will be used to let user know that we are in the progress of redefining classes.

Comment (yaohaizh) [reply src/hotCodeReplace.ts]: Okay, we need refine the event types for HCR.  

Currently, this event current only for error handling on the user side.

For hotCodeReplacestart/hotCodeReplaceEnd we need first add them on the debug server side.

Comment (yaohaizh) [reply src/hotCodeReplace.ts]: Refined HCR event.

**Pull request merged**